File: RevereString/reverse_string.js
let reverse_string = function (s) {
    let i= 0, j = s.length - 1;
    while (i < j) {
        let temp = s[i]
        s[i] = s[j]
        s[j] = temp
        i += 1
        j -= 1
    }
    return s
}

console.log(reverse_string([1,2,3,4,5]))

File: RevereString/reverse_string.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution{
    public:
        string reverse_string(string s){
            int i = 0, j = s.length() - 1;
            while(i < j){
                char c = s[i];
                s[i] = s[j];
                s[j] = c;
                i++;
                j--;
            }
            return s;
        }
};


int main(){
    Solution s;
    string str = "abcdef";
    cout << s.reverse_string(str) << endl;
}